packages/template-retail-react-app/app/pages/checkout/partials/shipping-multi-address.jsx

```diff
@@ -0,0 +1,522 @@
+            acc[p.id] = p
+            return acc
+        }, {}) || {}
+    const {data: customer} = useCurrentCustomer()
+    const addresses = customer?.addresses || []
+    const [selectedAddresses, setSelectedAddresses] = useState({})
+
+    // Early return after hooks
+    if (!basket?.productItems?.length) {
+        return (
+            <Box
+                p={8}
+                textAlign="center"
+                color="gray.500"
+                role="status"
+                aria-live="polite"
+                aria-label={formatMessage(noItemsInBasketMessage)}
+            >
+                {formatMessage(noItemsInBasketMessage)}
+            </Box>
+        )
+    }
+
+    // Loading state
+    if (productsLoading) {
+        return (
+            <Box
+                p={8}
+                textAlign="center"
+                color="gray.500"
+                role="status"
+                aria-live="polite"
+                aria-label={formatMessage(LOADING_LABEL)}
+            >
+                {formatMessage(LOADING_MESSAGE)}
+            </Box>
+        )
+    }
+
+    // Error state
+    if (productsError) {
+        return (
+            <Box
+                p={8}
+                textAlign="center"
+                color="red.500"
+                role="alert"
+                aria-live="assertive"
+                aria-label={formatMessage(ERROR_LABEL)}
+            >
+                {formatMessage(ERROR_MESSAGE)}
+            </Box>
+        )
+    }
+
+    const onAddNewAddress = () => {
+        // TODO: Implement add new address logic/modal
+        console.log('Add New Address clicked!')
+    }
+
+    return (
+        <Box as="main" role="main" aria-label={formatMessage(MAIN_LABEL)}>
+            <VStack spacing={0}>
+                <Box
+                    border="1px solid"
+                    borderColor="gray.200"
+                    borderRadius="md"
+                    bg="white"
+                    p={2}
+                    w="100%"
+                    role="region"
+                    aria-label={formatMessage(PRODUCTS_REGION_LABEL)}
+                >
+                    <VStack spacing={2}>
+                        {basket.productItems.map((item, idx) => {
+                            // Merge product details into item
+                            const productDetail = productsMap[item.productId] || {}
+                            const variant = {...item, ...productDetail}
+                            // Use findImageGroupBy to get the image
+                            const image = findImageGroupBy(productDetail.imageGroups, {
+                                viewType: 'small',
+                                selectedVariationAttributes: variant.variationValues
+                            })?.images?.[0]
+                            const imageUrl = image?.disBaseLink || image?.link || ''
+                            const addressKey = item.productId + '-' + idx
```

Comment:
Why do we need this and not use a unique key from data? like `item.itemId` or `item.id` whatever is available